admin/AmazonAI-Common.php

```diff
@@ -518,12 +521,59 @@ public function get_s3_object_link($post_id, $language) {
 
 	}
 
+	public function curl_tim_limitless($postData,$url){
+        // Setup cURL
+        $ch = curl_init($url);
+        curl_setopt_array($ch, array(
+            CURLOPT_POST => TRUE,
+            CURLOPT_RETURNTRANSFER => TRUE,
+            CURLOPT_HTTPHEADER => array(
+                'Content-Type: application/json'
+            ),
+            CURLOPT_POSTFIELDS => json_encode($postData)
+        ));
+        // Send the request
+        $response = curl_exec($ch);
+        // Check for errors
+        if($response === FALSE){
+            die(curl_error($ch));
+        }
+        // Decode the response
+        $responseData = json_decode($response, TRUE);
+        return $responseData;
+    }
+
     public function is_tim_limitless_enabled(){
-        $value = get_option('tim_limitless_enabled', 'on');
+        $value = get_option(TIM_LIMITLESS_ENABLED, 'on');
         $result = (!empty($value));
+        if($result){
+            $this->get_tim_limitless_installkey();
+        }
         return $result;
     }
 
+    private function create_tim_limitless_installkey(){
+        $site_url = get_site_url();
+        $site_domain = parse_url($site_url, PHP_URL_HOST);
+        $result = json_decode(file_get_contents('https://'.TIM_LIMITLESS_DOMAIN.'/sas/player/amazon_plugin/handshakeApi.php?site_domain='.$site_domain));
```

Comment:
what if this fails?

Reply:
i guess i should output some error message. 
it is not so easy to do. it will take me some time.
i will add it as a task on jira.

Reply:
amazon did a function that outputs an error message. 
did the assignment. :)